src/shims/unix/freebsd/foreign_items.rs

```diff
@@ -62,6 +63,15 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriInterpCxExt<'mir, 'tcx> {
                 this.gen_random(ptr, len)?;
                 this.write_scalar(Scalar::from_target_usize(len, this), dest)?;
             }
+            "ftruncate" => {
+                if this.tcx.sess.target.pointer_width == 32 {
```

Comment:
Does this function really only exist on 64bit FreeBSD? That's very strange...

Reply:
Oh I guess the problem is that the second argument is actually `off_t`, which I assume is `isize`, not `i64`?

having functions only exist on some bitwidths is too surprising IMO. Please change `ftruncate64` to take `fd: i32, length: i64` rather than taking `OpTy`. Then Linux and FreeBSD can share the implementation.